extension/js/taskdetails.js

```diff
@@ -3,99 +3,243 @@
 
         svg.find('.deserialization-time-proportion')
-            .attr('x', '' + metrics.deserializationTimeProportionPos + '%')
-            .attr('width', '' + metrics.deserializationTimeProportion + '%');
+            .attr('x', `${metrics.deserializationTimeProportionPos}%`)
+            .attr('width', `${metrics.deserializationTimeProportion}%`);
         e.find('.deserialization-time').text(moment.duration(metrics.deserializationTime).format(format));
 
         svg.find('.shuffle-read-time-proportion')
-            .attr('x', '' + metrics.shuffleReadTimeProportionPos + '%')
-            .attr('width', '' + metrics.shuffleReadTimeProportion + '%');
+            .attr('x', `${metrics.shuffleReadTimeProportionPos}%`)
+            .attr('width', `${metrics.shuffleReadTimeProportion}%`);
         e.find('.shuffle-read-time').text(moment.duration(metrics.shuffleReadTime).format(format));
 
         svg.find('.executor-runtime-proportion')
-            .attr('x', '' + metrics.executorComputingTimeProportionPos + '%')
-            .attr('width', '' + metrics.executorComputingTimeProportion + '%');
+            .attr('x', `${metrics.executorComputingTimeProportionPos}%`)
+            .attr('width', `${metrics.executorComputingTimeProportion}%`);
         e.find('.executor-runtime').text(moment.duration(metrics.executorComputingTime).format(format));
 
         svg.find('.shuffle-write-time-proportion')
-            .attr('x', '' + metrics.shuffleWriteTimeProportionPos + '%')
-            .attr('width', '' + metrics.shuffleWriteTimeProportion + '%');
+            .attr('x', `${metrics.shuffleWriteTimeProportionPos}%`)
+            .attr('width', `${metrics.shuffleWriteTimeProportion}%`);
         e.find('.shuffle-write-time').text(moment.duration(metrics.shuffleWriteTime).format(format));
 
         svg.find('.serialization-time-proportion')
-            .attr('x', '' + metrics.serializationTimeProportionPos + '%')
-            .attr('width', '' + metrics.serializationTimeProportion + '%');
+            .attr('x', `${metrics.serializationTimeProportionPos}%`)
+            .attr('width', `${metrics.serializationTimeProportion}%`);
         e.find('.serialization-time').text(moment.duration(metrics.serializationTime).format(format));
 
         svg.find('.getting-result-time-proportion')
-            .attr('x', '' + metrics.gettingResultTimeProportionPos + '%')
-            .attr('width', '' + metrics.gettingResultTimeProportion + '%');
+            .attr('x', `${metrics.gettingResultTimeProportionPos}%`)
+            .attr('width', `${metrics.gettingResultTimeProportion}%`);
         e.find('.getting-result-time').text(moment.duration(metrics.gettingResultTime).format(format));
     }
 }
 
+/**
+ * Shows a popup dialog with details of a task.
+ * @param {Object} item - data about the task.
+ */
+function showTaskDetails(item) {
+    const taskHTML = `
```

Comment:
One of the things I'm unsure about is how I've replaced the html import with a string. I do not think this is the best solution but, from what I can tell, importing HTML requires a custom loader in a webpack config and it seems (?) like JupyterLab doesn't support custom webpack configs.

Reply:
I decided to write this code inline since it was generally quite repetitive